torch/nn/functional.py

```diff
@@ -531,3 +531,20 @@ def upsample_bilinear(input, size=None, scale_factor=None):
         scale_factor (int): multiplier for spatial size. Has to be an integer.
     """
     return _functions.thnn.UpsamplingBilinear2d(size, scale_factor)(input)
+
+def cosine_similarity(input, eps=1e-12):
```

Comment:
I think it's nicer to accept two inputs in here. That's the convention we have in all other parts of the API